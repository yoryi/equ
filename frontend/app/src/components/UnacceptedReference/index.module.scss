@import "src/styles/mixins/breakpoints";

:local(.container) {
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  :local(.profileContainer) {
    display: flex;
    flex-direction: row;
    align-items: center;
    flex: 1 1 0;
    min-width: 0;

    :local(.detailsContainer) {
      flex: 1 1 0;
      min-width: 0;

      :local(.name) {
        overflow-wrap: break-word;
        text-overflow: ellipsis;
        white-space: nowrap;
        overflow: hidden;
      }
  
      :local(.position),
      :local(.classes) {
        font-family: Ilisarniq, sans-serif;
        font-size: 16px;
        line-height: 24px;
        color: var(--grey);
  
        @include breakpoint('small') {
          font-size: 12px;
          line-height: 18px;
        }
      }
    }
  }

  :local(.buttonContainer) {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
}
