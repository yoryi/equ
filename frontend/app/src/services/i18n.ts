import filesize from "filesize"
import i18n from "i18next"
import _ from "lodash"
import { initReactI18next } from "react-i18next"

import { en } from "../translations"

void i18n.use(initReactI18next).init({
  ns: [`common`, `screens`],
  defaultNS: `screens`,
  resources: {
    en,
  },
  lng: `en`,
  fallbackLng: `en`,
  interpolation: {
    format: (value, format, lng) => {
      if (!format) {
        return value
      }

      return format.split(`,`).reduce((value: string, format: string) => {
        switch (format.trim()) {
          case `localeString`:
            return value.toLocaleString()
          case `enIndefiniteArticle`:
          case `enIndefiniteArticleCapitalized`: {
            if (!lng || lng !== `en`) {
              return value
            }

            const letters = [`a`, `e`, `i`, `o`, `u`, `h`]
            const firstLetter = value.substring(0, 1)
            let article = ``
            if (
              letters.find((letter) => firstLetter.toLowerCase() === letter)
            ) {
              article =
                format === `enIndefiniteArticleCapitalized` ? `An` : `an`
            } else {
              article = format === `enIndefiniteArticleCapitalized` ? `A` : `a`
            }

            return `${article} ${value}`
          }
          case `title`: {
            return value
              .toLowerCase()
              .split(` `)
              .map((word: string) => _.upperFirst(word))
              .join(` `)
          }
          case `lowerCase`: {
            return value.toLowerCase()
          }
          case `fileSize`: {
            return filesize(parseInt(value), { round: 0 })
          }
          default:
            return value
        }
      }, value)
    },
  },
})

export default i18n
