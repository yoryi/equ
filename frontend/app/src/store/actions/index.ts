export * from "./admin"
export * from "./attachments"
export * from "./auth"
export * from "./follow"
export * from "./help"
export * from "./journeys"
export * from "./media"
export * from "./notifications"
export * from "./profile"
export * from "./searchAndFilters"
export * from "./settings"
export * from "./shared"
export * from "./student"
export * from "./surveys"
export * from "./university"
