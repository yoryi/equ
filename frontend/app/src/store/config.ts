interface Config {
  baseURL: string | null
}

export const config: Config = {
  baseURL: null,
}

export const initReduxStore = (baseURL: string) => {
  config.baseURL = baseURL
}
