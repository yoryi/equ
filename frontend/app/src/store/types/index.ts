export * from "./admin"
export * from "./auth"
export * from "./errors"
export * from "./ethnicity"
export * from "./follow"
export * from "./gender"
export * from "./hardship"
export * from "./help"
export * from "./highSchool"
export * from "./interest"
export * from "./notifications"
export * from "./profile"
export * from "./race"
export * from "./search"
export * from "./settings"
export * from "./shared"
export * from "./store"
export * from "./student"
export * from "./survey"
export * from "./university"
export * from "./utils"
