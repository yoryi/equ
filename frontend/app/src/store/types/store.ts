import { AdminReducer } from "../reducers/admin/reducer"
import { AuthReducer } from "../reducers/auth/reducer"
import { FollowReducer } from "../reducers/follow"
import { MediaReducer } from "../reducers/media/reducer"
import { NotificationsReducer } from "../reducers/notifications"
import { ProfileReducer } from "../reducers/profile/reducer"
import { SearchAndFiltersReducer } from "../reducers/searchAndFilters/reducer"
import { SettingsReducer } from "../reducers/settings/reducer"
import { SharedReducer } from "../reducers/shared/reducer"
import { StudentReducer } from "../reducers/student"
import { SurveysReducer } from "../reducers/surveys/reducer"
import { UniversityReducer } from "../reducers/university"

export interface ReduxState {
  auth: AuthReducer
  media: MediaReducer
  notifications: NotificationsReducer
  profile: ProfileReducer
  shared: SharedReducer
  settings: SettingsReducer
  student: StudentReducer
  surveys: SurveysReducer
  university: UniversityReducer
  follow: FollowReducer
  searchAndFilters: SearchAndFiltersReducer
  admin: AdminReducer
}
