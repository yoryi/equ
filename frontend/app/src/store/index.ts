import * as actions from "./actions"
import { initReduxStore } from "./config"
import { store } from "./createStore"

export { actions,initReduxStore, store }
