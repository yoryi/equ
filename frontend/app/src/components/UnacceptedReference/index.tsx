import * as mui from "@material-ui/core"
import classNames from "classnames"
import * as React from "react"
import { useTranslation } from "react-i18next"

import { ReactComponent as Eye } from "../../assets/eye.svg"
import { Reference } from "../../store/types"
import Button from "../Button/Button"
import ProfilePhoto from "../ProfilePhoto"
import RoundButton from "../RoundButton"
import Styles from "./index.module.scss"

interface UnacceptedReferenceBaseProps {
  className?: string
}

interface UnacceptedReferenceProps {
  reference: Reference
  onClick?: () => void
  loading?: undefined
}

interface LoadingUnacceptedReferenceProps {
  reference?: undefined
  onClick?: undefined
  loading?: true
}

const UnacceptedReference: React.FC<
  UnacceptedReferenceBaseProps &
    (UnacceptedReferenceProps | LoadingUnacceptedReferenceProps)
> = ({ className, reference, onClick, loading }) => {
  const { t } = useTranslation()

  return (
    <div className={classNames(Styles.container, className)}>
      <div className={Styles.profileContainer}>
        {reference && <ProfilePhoto reference={reference} />}
        {loading && <ProfilePhoto loading />}

        <div
          className={classNames(
            Styles.detailsContainer,
            `d-flex flex-column mx-3`,
          )}
        >
          <h3 className={Styles.name}>
            {!loading && `${reference?.firstName} ${reference?.lastName}`}
            {loading && <mui.Skeleton width={200} />}
          </h3>

          <span className={Styles.position}>
            {!loading && reference?.position}
            {loading && <mui.Skeleton width={100} />}
          </span>

          <span className={Styles.classes}>
            {!loading && reference?.organization}
            {loading && <mui.Skeleton width={125} />}
          </span>
        </div>
      </div>

      <div className={Styles.buttonContainer}>
        {!loading && (
          <>
            <Button
              className={`d-none d-md-block`}
              size={`xs`}
              onClick={onClick}
            >
              {t(`notificationsScreen.review`)}
            </Button>

            <RoundButton
              variant={`blue`}
              className={`d-block d-md-none`}
              onClick={onClick}
            >
              <Eye />
            </RoundButton>
          </>
        )}

        {loading && <mui.Skeleton width={40} height={40} variant="circular" />}
      </div>
    </div>
  )
}

export default UnacceptedReference
