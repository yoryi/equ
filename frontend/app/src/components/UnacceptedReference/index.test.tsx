//Testing
import { render } from "@testing-library/react"
import faker from "faker"
import * as React from "react"
//Translations
import { I18nextProvider } from "react-i18next"
//Redux store
import { Provider } from "react-redux"
import configureStore from "redux-mock-store"

import i18n from "../../services/i18n"
//Types
import { ReduxState, Reference, ReferenceType } from "../../store/types"
//Components
import UnacceptedReference from "./"

type TestStore = Pick<ReduxState, "media">

describe(`UnacceptedReference component`, () => {
  faker.seed(1000)

  const mockStore = configureStore<TestStore>()
  const store = mockStore({
    media: {
      mediaToken: faker.random.alphaNumeric(32),
    },
  })

  let reference: Reference
  beforeEach(() => {
    reference = {
      id: faker.random.number(10000),
      type: ReferenceType.Academic,
      firstName: faker.name.firstName(),
      lastName: faker.name.lastName(),
      media: null,
      organization: faker.company.companyName(),
      position: faker.lorem.word(),
      content: faker.lorem.sentences(3),
      isVisible: false,
      referenceLink: {
        id: faker.random.number(10000),
        type: ReferenceType.Academic,
        activity: null,
        token: faker.random.alphaNumeric(32),
        isActive: true,
        createdAt: null,
      },
    }
  })

  const wrapper: React.FC = ({ children }) => (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>{children}</Provider>
    </I18nextProvider>
  )

  describe(`should render and match the snapshot`, () => {
    it(`not loading`, () => {
      const { container } = render(
        <UnacceptedReference reference={reference} />,
        { wrapper },
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    it(`loading`, () => {
      const { container } = render(<UnacceptedReference loading />, {
        wrapper,
      })

      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
