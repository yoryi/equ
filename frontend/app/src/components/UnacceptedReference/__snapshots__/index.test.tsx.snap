// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnacceptedReference component should render and match the snapshot loading 1`] = `
<div
  class="container"
>
  <div
    class="profileContainer"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse rounded-circle css-fomgco-MuiSkeleton-root"
      style="width: 64px; height: 64px;"
    />
    <div
      class="detailsContainer d-flex flex-column mx-3"
    >
      <h3
        class="name"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style="width: 200px;"
        />
      </h3>
      <span
        class="position"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style="width: 100px;"
        />
      </span>
      <span
        class="classes"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style="width: 125px;"
        />
      </span>
    </div>
  </div>
  <div
    class="buttonContainer"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-fomgco-MuiSkeleton-root"
      style="width: 40px; height: 40px;"
    />
  </div>
</div>
`;

exports[`UnacceptedReference component should render and match the snapshot not loading 1`] = `
<div
  class="container"
>
  <div
    class="profileContainer"
  >
    <div
      class="container"
    >
      <span
        class="initials"
      >
        AW
      </span>
    </div>
    <div
      class="detailsContainer d-flex flex-column mx-3"
    >
      <h3
        class="name"
      >
        Aryanna Wilderman
      </h3>
      <span
        class="position"
      >
        expedita
      </span>
      <span
        class="classes"
      >
        Waters, Flatley and Jerde
      </span>
    </div>
  </div>
  <div
    class="buttonContainer"
  >
    <button
      class="primary-button d-none d-md-block default-button xs-button blue-button"
    >
      Review
    </button>
    <button
      class="button blue d-block d-md-none"
    >
      <div
        class="content loader exited"
      >
        <div
          aria-busy="true"
          class=""
        >
          <svg
            aria-label="audio-loading"
            fill="white"
            height="24"
            viewBox="0 0 120 30"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="15"
              cy="15"
              r="15"
            >
              <animate
                attributeName="r"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="15"
                repeatCount="indefinite"
                to="15"
                values="15;9;15"
              />
              <animate
                attributeName="fillOpacity"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="1"
                repeatCount="indefinite"
                to="1"
                values="1;.5;1"
              />
            </circle>
            <circle
              attributeName="fillOpacity"
              cx="60"
              cy="15"
              from="1"
              r="9"
              to="0.3"
            >
              <animate
                attributeName="r"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="9"
                repeatCount="indefinite"
                to="9"
                values="9;15;9"
              />
              <animate
                attributeName="fillOpacity"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="0.5"
                repeatCount="indefinite"
                to="0.5"
                values=".5;1;.5"
              />
            </circle>
            <circle
              cx="105"
              cy="15"
              r="15"
            >
              <animate
                attributeName="r"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="15"
                repeatCount="indefinite"
                to="15"
                values="15;9;15"
              />
              <animate
                attributeName="fillOpacity"
                begin="0s"
                calcMode="linear"
                dur="0.8s"
                from="1"
                repeatCount="indefinite"
                to="1"
                values="1;.5;1"
              />
            </circle>
          </svg>
        </div>
      </div>
      <div
        class="content entered"
      >
        <svg>
          eye.svg
        </svg>
      </div>
      <div
        class="content success exited"
      >
        <svg>
          check.svg
        </svg>
      </div>
      <div
        class="content exited"
      >
        <svg>
          close-icon-white.svg
        </svg>
      </div>
    </button>
  </div>
</div>
`;
